### PR TITLE
migrate tests from using HudsonTestCase to JenkinsRule

### PR DESCRIPTION
**Pull request by darxriggs**
- these are simple tests without Jelly files or actions implemented in the test class

### CODE REVIEW

File: test/src/test/groovy/hudson/cli/EnableJobCommandTest.groovy [hunk 1]
```diff
@@ -42,8 +50,7 @@ public class EnableJobCommandTest extends HudsonTestCase {
             cli.execute(["enable-job",p.name])
             assertFalse(p.disabled)
         } finally {
-            cli.close();
+            cli.close()
```
Comment (darxriggs) [inline test/src/test/groovy/hudson/cli/EnableJobCommandTest.groovy]: Shame on me. I always try to be as focused as possible because I know how hard it is to review tons of unrelated changes.

File: test/src/test/java/hudson/model/UsageStatisticsTest.java [hunk 2]
```diff
@@ -40,11 +44,16 @@
 /**
  * @author Kohsuke Kawaguchi
  */
-public class UsageStatisticsTest extends HudsonTestCase {
+public class UsageStatisticsTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
```
Comment (darxriggs) [inline test/src/test/java/hudson/model/UsageStatisticsTest.java]: Unfortunately the JenkinsRule is required even if not obvious at first sight.

``UsageStatistics`` accesses the Jenkins instance singleton internally.

File: test/src/test/java/jenkins/security/ApiTokenPropertyTest.java [hunk 3]
```diff
@@ -50,7 +63,7 @@ public Credentials getCredentials(AuthScheme scheme, String host, int port, bool
             protected HttpClient getHttpClient() {
                 HttpClient c = super.getHttpClient();
                 c.getParams().setAuthenticationPreemptive(true);
-                c.getState().setCredentials(new AuthScope("localhost", localPort, AuthScope.ANY_REALM), new UsernamePasswordCredentials("foo", token));
+                c.getState().setCredentials(new AuthScope("localhost", AuthScope.ANY_PORT, AuthScope.ANY_REALM), new UsernamePasswordCredentials("foo", token));
```
Comment (darxriggs) [inline test/src/test/java/jenkins/security/ApiTokenPropertyTest.java]: Because ``localPort`` is protected in ``JenkinsRule``. So accessing it would have required further changes.

Restricting the ``AuthScope`` that much for testing is not required. Even ``AuthScope.ANY`` would be fine. Anything else I would only use for security reasons on production not to provide credentials to an untrusted server by mistake.

File: test/src/test/java/hudson/model/UsageStatisticsTest.java (hunk 2 above)
Comment (jglick) [inline test/src/test/java/hudson/model/UsageStatisticsTest.java]: I suspected something like that. Another common case is code which calls `Secret`, which requires `Jenkins.instance` for the master keystore.

File: test/src/test/java/jenkins/security/ApiTokenPropertyTest.java (hunk 3 above)
Comment (jglick) [inline test/src/test/java/jenkins/security/ApiTokenPropertyTest.java]: > `localPort` is protected in `JenkinsRule`

I think it could be made `public`.
